docs/source/chainermn/model_parallel/overview.rst

```diff
@@ -0,0 +1,62 @@
+Overview
+========
+
+Model Parallelism
+~~~~~~~~~~~~~~~~~
+
+Even though ChainerMN mainly supports the data parallel approach for distributed training, it also has experimental APIs for the *model parallel* approach.
+The model parallel approach splits a given model into subcomponents loaded on several processes.
+This approach is useful in cases where
+
+- large mini-batch or high-resolusion is needed.
+- the model is too huge to run on a single process.
+- the mixture of experts are trained.
+
+.. figure:: ../../../image/parallelism.png
+    :align: center
+
+
+Philosophy
+~~~~~~~~~~
+
+ChainerMN takes the following three approaches to realize the model parallelism.
+
+1. Communication as Function
+----------------------------
+
+ChainerMN provides several special functions for communications such as ``chainermn.functions.bcast`` and ``chainermn.functions.alltoall``, which wraps raw MPI communications.
+Users define communications between processes as Chainer function calls in the model definitions.
+This enables highly flexible communication patterns.
+Moreover, parameter updates in backward propagation are automatically invoked through ``backward`` defined in those functions for communications.
+
+.. figure:: ../../../image/model_parallel/communication_as_function.png
+    :align: center
+    :scale: 40%
+
+
+2. Synchronous Model Parallel
+-----------------------------
+
+ChainerMN restricts itself to the synchronous updates.
+Though the asynchronous counterpart seems to be more computationally efficient, the asynchronous updates often suffer from the stale gradients problem and results in difficulty while debugging.
```

Comment:
the synchronous updates -> synchronous updates
the async... updates -> async updates

I think it'd better to user terminology of "synchronous SGD" and "asynchronous SGD", finishing with last sentence ChainerMN's synchronous communication model makes SGD simpler.